File: InheritanceBangunDatar/BDatar.cpp
#include <iostream>

// Class Induk: BangunDatar
class BangunDatar {
public:
    virtual double HitungLuas() = 0; // Fungsi untuk menghitung luas
    virtual double HitungKeliling() = 0; // Fungsi untuk menghitung keliling
};

// Class Anak: Persegi (Inheritance dari BangunDatar)
class Persegi : public BangunDatar {
private:
    double sisi;

public:
    Persegi(double s) : sisi(s) {}

    double HitungLuas() override {
        return sisi * sisi;
    }
    double HitungKeliling() override {
        return 4 * sisi;
    }
};

// Class Anak: Persegi Panjang (Inheritance dari BangunDatar)
class PersegiPanjang : public BangunDatar {
private:
    double panjang;
    double lebar;

public:
    PersegiPanjang(double p, double l) : panjang(p), lebar(l) {}

    double HitungLuas() override {
        return panjang * lebar;
    }
    double HitungKeliling() override {
        return 2 * (panjang + lebar);
    }
};

// Class Anak: Segitiga (Inheritance dari BangunDatar)
class Segitiga : public BangunDatar {
private:
    double alas;
    double tinggi;
    double sisi2;
    double sisi3;

public:
    Segitiga(double a, double t, double s2, double s3) : alas(a), tinggi(t), sisi2(s2), sisi3(s3) {}

    double HitungLuas() override {
        return 0.5 * alas * tinggi;
    }

    double HitungKeliling() override {
        return alas + sisi2 + sisi3;
    }
};

// Class Anak: Lingkaran (Inheritance dari BangunDatar)
class Lingkaran : public BangunDatar {
private:
    double jariJari;

public:
    Lingkaran(double r) : jariJari(r) {}

    double HitungLuas() override {
        return 3.14 * jariJari * jariJari;
    }

    double HitungKeliling() override {
        return 2 * 3.14 * jariJari;
    }
};

int main() {
    Persegi persegi(5.0);
    PersegiPanjang persegipanjang(4.0, 6.0);
    Segitiga segitiga(3.0, 4.0, 5.0, 2.0);
    Lingkaran lingkaran(2.5);

    std::cout << "Luas dan Keliling Persegi:" << std::endl;
    std::cout << "Luas: " << persegi.HitungLuas() << std::endl;
    std::cout << "Keliling: " << persegi.HitungKeliling() << std::endl;

    std::cout << "\nLuas dan Keliling Persegi Panjang:" << std::endl;
    std::cout << "Luas: " << persegipanjang.HitungLuas() << std::endl;
    std::cout << "Keliling: " << persegipanjang.HitungKeliling() << std::endl;

    std::cout << "\nLuas dan Keliling Segitiga:" << std::endl;
    std::cout << "Luas: " << segitiga.HitungLuas() << std::endl;
    std::cout << "Keliling: " << segitiga.HitungKeliling() << std::endl;

    std::cout << "\nLuas dan Keliling Lingkaran:" << std::endl;
    std::cout << "Luas: " << lingkaran.HitungLuas() << std::endl;
    std::cout << "Keliling: " << lingkaran.HitungKeliling() << std::endl;

    return 0;
}

File: InheritanceBangunDatar/BDatar.php
<?php

// Class parent (superclass) "BangunDatar"
class BangunDatar {
    // Properties
    protected $luas;
    protected $keliling;

    // Constructor
    public function __construct($luas, $keliling) {
        $this->luas = $luas;
        $this->keliling = $keliling;
    }

    // Method untuk menghitung luas
    public function hitungLuas() {
        return $this->luas;
    }

    // Method untuk menghitung keliling
    public function hitungKeliling() {
        return $this->keliling;
    }
}

// Class child "Persegi" (inherit dari BangunDatar)
class Persegi extends BangunDatar {
    // Constructor
    public function __construct($sisi) {
        $luas = $sisi * $sisi;
        $keliling = 4 * $sisi;
        parent::__construct($luas, $keliling);
    }
}

// Class child "PersegiPanjang" (inherit dari BangunDatar)
class PersegiPanjang extends BangunDatar {
    // Constructor
    public function __construct($panjang, $lebar) {
        $luas = $panjang * $lebar;
        $keliling = 2 * ($panjang + $lebar);
        parent::__construct($luas, $keliling);
    }
}

// Class child "Segitiga" (inherit dari BangunDatar)
class Segitiga extends BangunDatar {
    // Constructor
    public function __construct($alas, $tinggi, $sisi2, $sisi3) {
        $luas = 0.5 * $alas * $tinggi;
        $keliling = $alas + $sisi2 + $sisi3;
        parent::__construct($luas, $keliling);
    }
}

// Class child "Lingkaran" (inherit dari BangunDatar)
class Lingkaran extends BangunDatar {
    // Constructor
    public function __construct($jarijari) {
        $luas = M_PI * $jarijari * $jarijari;
        $keliling = 2 * M_PI * $jarijari;
        parent::__construct($luas, $keliling);
    }
}

// Function untuk menampilkan menu pilihan
function tampilkanMenu() {
    echo "Pilihan bangun datar:\n";
    echo "1. Persegi\n";
    echo "2. Persegi Panjang\n";
    echo "3. Segitiga\n";
    echo "4. Lingkaran\n";
}

// Contoh penggunaan
tampilkanMenu();
$choice = readline("Masukkan pilihan Anda: ");
switch ($choice) {
    case 1:
        $sisi = readline("Masukkan sisi Persegi: ");
        $persegi = new Persegi($sisi);
        echo "Luas Persegi: " . $persegi->hitungLuas() . " Keliling: " . $persegi->hitungKeliling() . "\n";
        break;
    case 2:
        $panjang = readline("Masukkan panjang Persegi Panjang: ");
        $lebar = readline("Masukkan lebar Persegi Panjang: ");
        $persegiPanjang = new PersegiPanjang($panjang, $lebar);
        echo "Luas Persegi Panjang: " . $persegiPanjang->hitungLuas() . " Keliling: " . $persegiPanjang->hitungKeliling() . "\n";
        break;
    case 3:
        $alas = readline("Masukkan alas atau sisi kesatu segitiga: ");
        $tinggi = readline("Masukkan tinggi Segitiga: ");
        $sisi2 = readline("Masukkan panjang sisi kedua Segitiga: ");
        $sisi3 = readline("Masukkan panjang sisi ketiga Segitiga: ");
        $segitiga = new Segitiga($alas, $tinggi, $sisi2, $sisi3);
        echo "Luas Segitiga: " . $segitiga->hitungLuas() . " Keliling: " . $segitiga->hitungKeliling() . "\n";
        break;
    case 4:
        $jarijari = readline("Masukkan jari-jari Lingkaran: ");
        $lingkaran = new Lingkaran($jarijari);
        echo "Luas Lingkaran: " . $lingkaran->hitungLuas() . " Keliling: " . $lingkaran->hitungKeliling() . "\n";
        break;
    default:
        echo "Pilihan tidak valid.\n";
}